extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/test/java/io/quarkus/resteasy/reactive/links/deployment/TestResource.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkus.resteasy.reactive.links.deployment;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import io.quarkus.resteasy.reactive.links.InjectRestLinks;
+import io.quarkus.resteasy.reactive.links.RestLink;
+import io.quarkus.resteasy.reactive.links.RestLinkType;
+
+@Path("/")
+public class TestResource {
+
+    private static final AtomicInteger ID_COUNTER = new AtomicInteger(0);
+
+    private static final List<TestRecord> RECORDS = new LinkedList<>(Arrays.asList(
+            new TestRecord(ID_COUNTER.incrementAndGet(), "First"),
+            new TestRecord(ID_COUNTER.incrementAndGet(), "Second")));
+
+    @GET
+    @Path("/with-links/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RestLink
+    @InjectRestLinks(RestLinkType.INSTANCE)
+    public TestRecord getById(@PathParam("id") int id) {
+        for (TestRecord record : RECORDS) {
+            if (record.getId() == id) {
+                return record;
+            }
+        }
+        throw new NotFoundException();
+    }
+
+    @GET
+    @Path("/with-links")
+    @Produces(MediaType.APPLICATION_JSON)
+    @RestLink(entityType = TestRecord.class, rel = "list")
```

Comment:
This should be optional, in which case we can infer that the entity type is `TestRecord` (from the return type `List<TestRecord>` and same for the `rel` which should be `collection`.

Reply:
I don't have this implemented yet, but inferring an entity type from a generic is in my todo. So for now this is a workaround.

Reply:
As for the rel, this annotation allows to define any rel you want.

Reply:
OK